tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -395,9 +398,353 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+			namespace, nil, scParameters, "", nil, "", true, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			ginkgo.By("Delete the SC created in step 2")
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(
+			client, []*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify that the created CNS volume is compliant and has correct policy id")
+		volumeID := pvs[0].Spec.CSI.VolumeHandle
+		storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyName)
+		e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+		gomega.Expect(e2eVSphere.verifyDatastoreMatch(volumeID, sharedvmfsURL)).To(
+			gomega.BeTrue(), "volume %v was created on wrong ds", sharedvmfsURL)
+
+		defer func() {
+			ginkgo.By("Delete the PVCs created in step 3")
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeID)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		}()
+
+		pvcs2d := [][]*v1.PersistentVolumeClaim{}
+		pvcs2d = append(pvcs2d, []*v1.PersistentVolumeClaim{pvclaim})
+		ginkgo.By("Create a pod using pvc1 say pod1 and wait for it to be ready")
+		pods := createMultiplePods(ctx, client, pvcs2d, true) // only 1 will be created here
+		defer func() {
+			ginkgo.By("Delete pod")
+			deletePodsAndWaitForVolsToDetach(ctx, client, pods, true)
+		}()
+
+		ginkgo.By("Get filesystem size for mount point /mnt/volume1 before expansion")
+		originalFsSize, err := getFSSizeMb(f, pods[0])
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Expand pvc1 to a large size this should take more than vpxd timeout")
+		currentPvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		newSize := currentPvcSize.DeepCopy()
+		newSize.Add(resource.MustParse(largeSize))
+		framework.Logf("currentPvcSize %v, newSize %v", currentPvcSize, newSize)
+		start := time.Now()
+		pvclaim, err = expandPVCSize(pvclaim, newSize, client)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+		err = waitForPvResize(pvs[0], client, newSize, totalResizeWaitPeriod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		elapsed := time.Since(start)
+
+		ginkgo.By("Verify PVC expansion took longer than vpxd timeout")
+		gomega.Expect(elapsed > time.Second*time.Duration(vpxdReducedTaskTimeoutSecsInt)).To(
+			gomega.BeTrue(), "PVC expansion was faster than vpxd timeout")
+
+		ginkgo.By("Waiting for file system resize to finish")
```

Comment:
Shouldn't this be before we calculate time elapsed as this is also part of PVC expansion?

Reply:
from VC side expansion is done before this point. File system resize wont result is extra space being reserved or allocated for the volume.